backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/ddorder/process/WEBUI_DD_OrderLine_MoveSelected_HU.java

```diff
@@ -0,0 +1,166 @@
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ui.web.ddorder.process;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.handlingunits.ddorder.api.IHUDDOrderBL;
+import de.metas.handlingunits.model.I_DD_OrderLine;
+import de.metas.handlingunits.model.I_M_HU;
+import de.metas.process.IProcessDefaultParameter;
+import de.metas.process.IProcessDefaultParametersProvider;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.ui.web.handlingunits.HUEditorRow;
+import de.metas.ui.web.handlingunits.HUEditorView;
+import de.metas.ui.web.handlingunits.WEBUI_HU_Constants;
+import de.metas.ui.web.process.adprocess.ViewBasedProcessTemplate;
+import de.metas.ui.web.view.IViewsRepository;
+import de.metas.ui.web.view.ViewId;
+import de.metas.ui.web.window.datatypes.DocumentIdsSelection;
+import de.metas.util.Services;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.SpringContextHolder;
+
+public class WEBUI_DD_OrderLine_MoveSelected_HU extends ViewBasedProcessTemplate implements IProcessPrecondition, IProcessDefaultParametersProvider
+{
+	private final IViewsRepository viewsRepository = SpringContextHolder.instance.getBean(IViewsRepository.class);
+	private final IHUDDOrderBL huDDOrderBL = Services.get(IHUDDOrderBL.class);
+
+	private static final String PARAM_M_HU_ID = I_M_HU.COLUMNNAME_M_HU_ID;
+	@Param(parameterName = PARAM_M_HU_ID, mandatory = true)
+	private int mHuID;
+
+	private static final String PARAM_DD_ORDER_LINE_ID = I_DD_OrderLine.COLUMNNAME_DD_OrderLine_ID;
+	@Param(parameterName = PARAM_DD_ORDER_LINE_ID, mandatory = true)
+	private int ddOrderLineId;
+
+	private static final String PARAM_LOCATOR_TO_ID = I_DD_OrderLine.COLUMNNAME_M_LocatorTo_ID;
+	@Param(parameterName = PARAM_LOCATOR_TO_ID, mandatory = true)
+	private int paramLocatorToId;
+
+
+
+	public ProcessPreconditionsResolution checkPreconditionsApplicable()
+	{
+		if (getSelectedRowIds().isEmpty())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+		else if (getSelectedRowIds().isMoreThanOneDocumentId())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
+		}
+		else if (huEditorWasNotOpenedFromADDOrderLine())
+		{
+			return ProcessPreconditionsResolution
+					.rejectWithInternalReason("The process should only be available when the HUEditor was opened from a DD_OrderLine window!");
+		}
+		else if (selectedRowIsNotATopHU())
+		{
+			return ProcessPreconditionsResolution
+					.reject(msgBL.getTranslatableMsgText(WEBUI_HU_Constants.MSG_WEBUI_ONLY_TOP_LEVEL_HU));
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+
+		final I_DD_OrderLine selectedDDOrderLine = InterfaceWrapperHelper.load(ddOrderLineId, I_DD_OrderLine.class);
```

Comment:
pls introduce `I_DD_OrderLine IDDOrderDAO.getLineById(int ddOrderLineId)` and use it here.
Avoid direct `InterfaceWrapperHelper.load` calls outside of repository/DAO.